src/dbg/xrefs.cpp

```diff
@@ -40,7 +40,7 @@ struct XrefSerializer : AddrInfoSerializer<XREFSINFO>
             record.addr = duint(json_hex_value(json_object_get(reference, "addr")));
             record.type = XREFTYPE(json_hex_value(json_object_get(reference, "type")));
             value.type = max(record.type, value.type);
-            value.references.insert({ record.addr, record });
+            value.references.emplace(record.addr, record);

```

Comment:
It's a matter of preference, but I prefer `insert` here and in the other places. There's some good reasoning here:
https://quuxplusone.github.io/blog/2021/03/03/push-back-emplace-back/

Other than that, looks good.

Reply:
Personally I tend to avoid following the melon man's advice. He makes a lot of 'technically correct' arguments and then actively suggests a _default_ that shouldn't apply unless you're a "library author" who has to consider every detail. A good example would be the [`for (auto&& elt : range)`](https://quuxplusone.github.io/blog/2018/12/15/autorefref-always-works/) post, which has likely ruined business-logic code for no good reason... `</rant>`

The linked post specifically talks about `vector`, where you can use `push_back` and `emplace_back` with the _same type_ and it then uses a (very questionable) benchmark to justify why you should always prefer `push_back`. The key difference here is that you pass the `w` directly. The correct way to use `emplace_back` would be: `widgets.emplace_back(foo, bar, baz);` (since constructs the object in-place), but he kind of glosses over this.

In our specific case I switched to `emplace` because it's simply more readable and I don't really see how the post about `vector` applies. It might if I did `references.emplace({ record.addr, record });` but this is not a valid use of `emplace` for map-like types...

Reply:
👍 Let's merge it, I think both are fine, I personally slightly prefer `insert` here, e.g. IDE type suggestions are usually better, but it's not that important.